src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -2608,5 +2608,11 @@ public void Base_05()
             }
             EOF();
         }
+
+        [Fact, WorkItem(51590, "https://github.com/dotnet/roslyn/issues/51590")]
+        public void ParseIncompleteRecordSyntax()
```

Comment:
I really like the idea here of essentially removing token, re typing and seeing if we can deal with it.

Did you consider making this a `[Theory]` and adding more cases here for testing

Reply:
Cool :-)
A `[Theory]` would make sense, but then I don't know where we should put it. Having a helper function makes it easy to call for new features.

Reply:
I was thinking of putting it right here. Imagine if the `[Theory]` here just had some data entries that enumerated a couple fo the common ways of defining records and making sure they could go through that tool.